api/kyverno/v1/image_verification_types.go

```diff
@@ -8,10 +8,25 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
+// ImageVerificationType selects the type of verification algorithm
+// +kubebuilder:validation:Enum=Cosign;NotaryV2
+// +kubebuilder:default=Cosign
+type ImageVerificationType string
+
+const (
+	Cosign   ImageVerificationType = "Cosign"
+	NotaryV2 ImageVerificationType = "NotaryV2"
+)
+
 // ImageVerification validates that images that match the specified pattern
 // are signed with the supplied public key. Once the image is verified it is
 // mutated to include the SHA digest retrieved during the registration.
 type ImageVerification struct {
+	// Type specifies the method of signature validation. The allowed options
+	// are Cosign and NotaryV2. By default Cosign is used if a type is not specified.
+	// +kubebuilder:validation:Optional
+	Type ImageVerificationType `json:"type,omitempty" yaml:"type,omitempty"`
```

Comment:
I'm still uncertain about how to handle v1 and v2beta1 types. Seems like the best is to add new fields to both, v1 is what's working currently and v2* is what we need in the future.